enumerable_methods.rb

```diff
@@ -0,0 +1,65 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    arr = []
+    my_each { |n| arr.push(n) if yield n } if block_given?
+    arr
+  end
```

Comment:
- [ ] enumerables #my_select returns an enumerator if no block is given
     Failure/Error: expect(array.my_select).to be_an(Enumerator)

Reply:
so I should just remove if block_given? condition?